api/proto/external.proto

```diff
@@ -423,6 +423,9 @@ message MintTxPrefix {
 
     /// The block index at which this transaction is no longer valid.
     uint64 tombstone_block = 6;
+
+    /// The (optional) encrypted fog hint of the minted TxOut.
+    EncryptedFogHint e_fog_hint = 7;
```

Comment:
One thing that made me uncomfortable when working on this is, I'm not sure that anything in the consensus enclave makes sure that ignored arguments in the mint command lead to an error.

So like, if someone used the new mint client (after this revision) to try to mint to a consensus network based on current prod (before this revision), I think the existing consensus enclaves would just strip out the e_fog_hint, and go on to mint without it. But that would not be what the user intended.

Maybe it would be a good idea to have the user specify a hash of the data they sent or something, and the consensus enclave checks it? This may avoid this issue in the future if we want to add more mint parameters.

There might be an easier way to make unknown fields an error, but I think usually protobuf doesn't allow that behavior.

Reply:
Good catch! Specifying a hash for the mint config seems like a good general strategy. Maybe with a block version bump? A BV bump also seems useful for gating usage of new fields.

Reply:
Since I think by definition all fields are optional in protobuf, the behavior on the Rust side depends on how Prost treats missing fields. Fields explicitly marked as `Option<>` are easy. If a field does not have the `required` modifier on the prost attribute then it will be defaulted (`Default::default()`)  and the enclave will likely not notice that. I am not sure how `required` behaves - according to [this](https://github.com/tokio-rs/prost#field-modifiers) table it is a `proto2` modifier and proto files are `proto3` so a little experiment to see if that results in an error when this is missing might be useful.
This will of course not work on scalar values, so the fool-proof solution is to include a hash.
I think we're at a point where making a breaking change to the relevant [propose](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/consensus/api/proto/consensus_client.proto#L76-L79) APIs is fine since it is seeing very little and controlled use, and we can make sure the minters and governors use the right client.

Reply:
Maybe we actually don't have to do anything here?

The minters sign the MintTx digest I think, so if there is something that the governors can see but the enclave cannot, then the minter signature checking will fail? So it may work fine in the scenario I'm imaginging (new minter software submits MintTx to old consensus node) for roughly the same reason that it works (and successfully rejects the Tx with unknown fields) in this scenario for a normal mobilecoin transactoin?

I want to trace the code to make sure my understanding is correct

Reply:
Yeah that's a good point...

Reply:
I think this is okay -- adding a field under the `MintTxPrefix` is going to affect that `MintTxPrefix` digest that is signed, so I think we don't have to do anything here